test/e2e/network/ingress.go

```diff
@@ -153,10 +155,49 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
 			// framework.ExpectNoError(jig.verifyURL(fmt.Sprintf("https://%v/", ip), "", 30, 1*time.Second, httpClient))
 		})
 
+		It("should create ingress with pre-shared certificate", func() {
+			preSharedCertName := "test-pre-shared-cert"
+			By(fmt.Sprintf("Creating ssl certificate %q on GCE", preSharedCertName))
+			testHostname := "test.ingress.com"
+			cert, key, err := framework.GenerateRSACerts(testHostname, true)
+			Expect(err).NotTo(HaveOccurred())
+			gceCloud, err := framework.GetGCECloud()
+			Expect(err).NotTo(HaveOccurred())
+			_, err = gceCloud.CreateSslCertificate(&compute.SslCertificate{
+				Name:        preSharedCertName,
+				Certificate: string(cert),
+				PrivateKey:  string(key),
+				Description: "pre-shared cert for ingress testing",
+			})
+			defer func() {
```

Comment:
I'd move the defer to before the CreateSslCertificate as long as you check for NotFound in the delete. It's a bit difficult to read having the create asserts so far away from the create.

Reply:
Make sense, done.